skalenetwork/skale-admin
# Exit on failed broadcast and alright transactions

Changes:
1. Updated to skale.py 6.1.x with fixed exception handling.
2. Do not swallow exceptions from broadcast and alright transactions.
3. Add last completed step to DKGResult.
4. Add ComplaintReason enum.
5. Add more DKG unit tests.
6. Switch to anvil to speed up overall test execution.
7. Update base image to python3.11.
8. Bump dependecies. 

Performance:
1. Performance should not be affected. 

Unit tests:
1. Added more unit tests in dkg_tests/main_test.py

Testing procedure:
1. Drain node's wallet before dkg.
2. Create schain. 
3. Make sure broadcast failed.
4. Refill wallet balance.
5. Make sure broacast is eventually sent and dkg completed.
